lib/gears/object.lua

```diff
@@ -122,9 +122,47 @@ function object:emit_signal(name, ...)
     end
 end
 
---- Returns a new object. You can call :emit_signal(), :disconnect_signal,
--- :connect_signal() and :add_signal() on the resulting object.
-local function new()
+local function get_miss(self, key)
+    local class = rawget(self, "_class")
+
+    if rawget(self, "get_"..key) then
+        return rawget(self, "get_"..key)(self)
+    elseif class and class["get_"..key] then
+        return class["get_"..key](self)
+    elseif class and class[key] then
+        return class[key]
+    else
+        return rawget(self, key)
```

Comment:
If we are in an `__index`-handler, this `rawget` will always return `nil`. Otherwise, the handler wouldn't be called in the first place. You can just remove this.